### PR TITLE
CI test cache fixes

### PR DESCRIPTION
**Pull request by supermario**
So I wasn't able to figure out what exactly was causing the problem, but it seems the ordering meant that somewhere likely a component in the test setup overall was compiling with vanilla Elm first, causing a package cache issue.

I think ultimately in Lamdera we'll have to figure out a clever way to detect/purge invalid caches on the fly, but for now at least this gets things green up to the cypress stage, where it seems to fail for unrelated reasons due to cypress upgrades that are beyond me 😄

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+set -ex;
+root=`pwd`
+yes | lamdera reset || true
```
Comment (supermario) [inline test.sh:3]: This `|| true` is a smell, I'll come back and remove it once I've done a release with the fix that led to this (edge case where reset fails if the folder it's trying to purge doesn't exist).

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -39,8 +39,6 @@ jobs:
       # Install npm packages, unless we restored them from cache.
       # Since `npm ci` removes the node_modules folder before running it’s
       # important to skip this step if cache was restored.
-      - name: Lamdera Reset for tests
-        run: rm -rf elm-stuff $HOME/.elm
```
Comment (supermario) [inline .github/workflows/ci.yml:43]: This is effectively the same as `lamdera reset`.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -68,9 +66,8 @@ jobs:
         run: npx --no-install elm-format --validate src/ generator/src
       - name: Build elm docs
         id: docs
-        run: npx --no-install elm make --docs docs.json
-      - name: Lamdera Reset for cypress
-        run: yes | lamdera reset
+        run: lamdera make --docs docs.json
```
Comment (supermario) [inline .github/workflows/ci.yml:69]: Should be no need to rest cache if we use Lamdera to do the docs.

**Comment by dillonkearns** [comment]
> I think ultimately in Lamdera we'll have to figure out a clever way to detect/purge invalid caches on the fly, but for now at least this gets things green up to the cypress stage, where it seems to fail for unrelated reasons due to cypress upgrades that are beyond me 😄

Oh yes, I'll take a look at the Cypress issues. Thanks so much for pushing this along Mario!

If there's a way to prevent the Elm and Lamdera caches from stepping on each others toes, that would avoid a lot of debugging for users, some of the people who are alpha testing elm-pages v3 have been struggling with this since they are maintaining apps where they use both Elm and lamdera (via elm-pages v3) for their daily workflows.
